cli/loader/src/lib.rs

```diff
@@ -220,6 +223,24 @@ impl Loader {
                     .and_then(|extension| {
                         self.language_configuration_ids_by_file_type.get(extension)
                     })
+            })
+            .or_else(|| {
+                let Ok(file) = fs::File::open(path) else {
+                    return None;
+                };
+                let reader = BufReader::new(file);
+                let Some(Ok(first_line)) = std::io::BufRead::lines(reader).next() else {
+                    return None;
+                };
+
+                self.language_configuration_ids_by_first_line_regex
+                    .iter()
+                    .find(|(regex, _)| {
+                        Regex::from_str(regex)
```

Comment:
See above comment, it's not a good idea to instantiate Regex instances here every time.